test/index.js

```diff
@@ -138,6 +138,39 @@ test('SubDb main function', function (t) {
       t.end()
     })
   })
+  t.test('iterator options are forwarded (issue #1)', function (t) {
+    t.plan(4)
+    var enc = { keyEncoding: 'utf8', valueEncoding: 'json' }
+    var db = levelup(encoding(memdown(), enc))
+    var sub = subdb(db, 'test', enc)
+    var key = 'foo'
+    var value = { hello: 'world' }
+    sub.once('open', function () {
+      sub.put(key, value, function () {
+        db.createReadStream().on('data', function (r) {
+          t.is(r.key, '!test!' + key, 'db key is utf8')
+          t.deepEqual(r.value, value, 'db value is json')
+        })
+        sub.createReadStream().on('data', function (r) {
+          t.is(r.key, key, 'sub db key is utf8')
+          t.deepEqual(r.value, value, 'db value is json')
+        })
+      })
+    })
+  })
+  t.test('concatenating Buffer keys', function (t) {
+    t.plan(1)
+    var db = levelup(memdown())
+    var sub = subdb(db, 'test', { keyEncoding: 'binary' })
+    var key = Buffer.from('foo')
```

Comment:
Put some non-utf8 bytes in there, to test that the key isn't inadvertently stringified. Same as uh, the other module we did this in, but I can't remember which ;)

Reply:
Good catch!